tools/src/blockchain-analyser/Types.hs

```diff
@@ -0,0 +1,74 @@
+{-# LANGUAGE TemplateHaskell       #-}
+{-# LANGUAGE TypeApplications      #-}
+{-# LANGUAGE TypeSynonymInstances  #-}
+
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE RecordWildCards       #-}
+
+module Types ( initBlockchainAnalyser
+             , DBFolderStat
+             , BlockchainInspector
+             , prevBlock
+             ) where
+
+import           Universum
+
+import           Control.Lens         (makeLensesWith)
+import qualified Control.Monad.Reader as Mtl
+import           Ether.Internal       (HasLens (..))
+import           Mockable             (Production)
+
+import           Pos.Block.Core       (Block, BlockHeader)
+import           Pos.Block.Types      (Undo)
+import           Pos.Core             (HasCoreConstants, HeaderHash, prevBlockL)
+import           Pos.DB               (MonadBlockDBGeneric (..), MonadDBRead (..))
+import qualified Pos.DB               as DB
+import qualified Pos.DB.Block         as BDB
+import           Pos.Ssc.GodTossing   (SscGodTossing)
+import           Pos.Util.Util        (postfixLFields)
+
+type DBFolderStat = (Text, Integer)
+
+-- | Enough context for analysing blocks.
+data BlockchainInspectorContext = BlockchainInspectorContext { tbgcNodeDBs :: DB.NodeDBs }
```

Comment:
I just noticed: `tbgc` prefix looks strange, usually we use abbreviation of type name as a prefix (so `bic` in this case).